### PR TITLE
NEW : Address/Contact by default on third parties

### PR DESCRIPTION
**Pull request by ghost**
NEW : Contact role

Implementation of contact role by default on proposal, order, invoice (customer and supplier) , event, project

### CODE REVIEW

**Comment by hregis** [comment]
@atm-florian hi Florian, warning with fk_soc constraint if contact not linked with company!

**Comment by ghost** [comment]
> @atm-florian hi Florian, warning with fk_soc constraint if contact not linked with company!

In this case you cannot define a role for a contact

**Comment by hregis** [comment]
@atm-florian ok je sors ! ;-)

File: htdocs/contact/card.php [hunk 1]
```diff
@@ -474,9 +475,6 @@
         // Si edition contact deja existant
         $object = new Contact($db);
         $res=$object->fetch($id, $user);
-        if ($res < 0) { dol_print_error($db, $object->error); exit; }
```
Comment (eldy) [inline htdocs/contact/card.php]: Why removing error reporting ?

Comment (atm-florian) [reply htdocs/contact/card.php]: True, initially I want to remove fetch_optional that is now into contact class fetch method. I've gone too far with this removing. Just send a commit with fix.

**Comment by atm-maxime** [comment]
@atm-florian, I hope you have a migration script to go from our module to this new standard :)

**Pull request merged**